### DISCUSSION TITLE
pac solution set-preferred

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/powerplatform-build-tools#819] [Ideas]
With the new [preferred solution feature](https://learn.microsoft.com/en-us/power-apps/maker/data-platform/preferred-solution), these commands would be a fantastic addition.

# Command

**pac solution set-preferred**

This command will enable you to set the preferred solution in the currently selected environment.

Usage: `pac solution set-preferred --name [--environment]`

| Parameter | Description |
|--------|--------|
| `--name` | 	The (internal) name of the solution you want to set as the preferred solution. (alias: -n) |
| `--environment` | 	(optional) The name of the environment in which you want to set the preferred solution. (alias: -env) |

**pac solution list-preferred**

This command will enable you to list the preferred solution. This will output the preferred solution name, internal name, and solution ID.

Usage: `pac solution list-preferred [--environment] [--json]`

| Parameter | Description |
|--------|--------|
| `--environment` | 	(optional) The name of the environment for which you want to list the preferred solution. (alias: -env) |
| `--json` | 	Willl enable you to return the output as JSON |

>> **Updated after comments by @appieschot**

**Comment by appieschot** [comment]
Do you expect a get and a delete command as well? If so a new subgroup (preferred) makes sense, so you can do a set, list and delete (or get) to retrieve / delete the current one?

**Reply by Laskewitz** [reply]
List would be a good addition as well. Delete isn't a thing with this I believe, because you can run the set command for that again to achieve that. It might be an option to set the preferred solution to the default solution when you omit the `--name` parameter. What do you think of that?

**Reply by appieschot** [reply]
If the docs explain that clearly that makes sense! 

From a UI perspective there is a delete option for the whole solution only, I missed that! 

so as a admin I expect to be able to unset whatever I would have set before. Setting it to the default might be the result, but that means I have to type the name of the default one, that might introduce risk with typos. So I agree that omitting the name parameters fall back to default makes sense.
